hydroflow_lang/src/graph/ops/difference_multiset.rs

```diff
@@ -0,0 +1,183 @@
+use quote::{quote_spanned, ToTokens};
+use syn::parse_quote;
+
+use super::{
+    DelayType, FlowProperties, FlowPropertyVal, OperatorCategory, OperatorConstraints,
+    OperatorWriteOutput, WriteContextArgs, RANGE_0, RANGE_1,
+};
+use crate::graph::{OperatorInstance, PortIndexValue};
+
+/// > 2 input streams of the same type T, 1 output stream of type T
+///
+/// For a given tick, forms the set difference of the items in the input
+/// streams, returning items in the `pos` input that are not found in the
+/// `neg` input.
+///
+/// `difference` can be provided with one or two generic lifetime persistence arguments
+/// in the same way as [`join`](#join), see [`join`'s documentation](#join) for more info.
+///
+/// ```hydroflow
+/// source_iter(vec!["dog", "cat", "elephant"]) -> [pos]diff;
+/// source_iter(vec!["dog", "cat", "gorilla"]) -> [neg]diff;
+/// diff = difference() -> assert_eq(["elephant"]);
+/// ```
+pub const DIFFERENCE_MULTISET: OperatorConstraints = OperatorConstraints {
+    name: "difference_multiset",
+    categories: &[OperatorCategory::MultiIn],
+    hard_range_inn: &(2..=2),
+    soft_range_inn: &(2..=2),
+    hard_range_out: RANGE_1,
+    soft_range_out: RANGE_1,
+    num_args: 0,
+    persistence_args: &(0..=2),
+    type_args: RANGE_0,
+    is_external_input: false,
+    ports_inn: Some(|| super::PortListSpec::Fixed(parse_quote! { pos, neg })),
+    ports_out: None,
+    properties: FlowProperties {
+        deterministic: FlowPropertyVal::Preserve,
+        monotonic: FlowPropertyVal::No,
+        inconsistency_tainted: false,
+    },
+    input_delaytype_fn: |idx| match idx {
+        PortIndexValue::Path(path) if "neg" == path.to_token_stream().to_string() => {
+            Some(DelayType::Stratum)
+        }
+        _else => None,
+    },
+    write_fn: |wc @ &WriteContextArgs {
+                           op_span,
+                           ident,
+                           inputs,
+                           op_inst:
+                               OperatorInstance {
+                                   ..
+                               },
+                           ..
+                       },
+                       diagnostics| {
+        let wc = WriteContextArgs {
```

Comment:
i don't think you need to construct a new wc since you aren't modifying anything in it. I think you can delete lines 59 to 62 here.

Reply:
Good catch. Also applied to `difference.rs` now.